src/general_manager/measurement/measurementField.py

```
@@ -1,168 +1,275 @@
         if value is None:
             setattr(instance, self.value_attr, None)
             setattr(instance, self.unit_attr, None)
             return
-        elif isinstance(value, str):
+        if isinstance(value, str):
             try:
                 value = Measurement.from_string(value)
             except ValueError:
                 raise ValidationError(
                     {self.name: ["Value must be a Measurement instance or None."]}
                 )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)